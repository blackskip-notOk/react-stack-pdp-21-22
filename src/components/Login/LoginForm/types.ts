export interface LoginFormData {
	email: string,
	password: string,
	rememberMe?: boolean,
	captcha?: string,
};

export interface LoginFormResult {
    email: { value: string },
    password: { value: string },
    rememberMe: { checked: boolean },
    captcha?: { value?: string },
}