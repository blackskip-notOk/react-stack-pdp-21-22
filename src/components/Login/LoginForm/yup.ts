import * as yup from "yup";

yup.setLocale({
    mixed: ({
        default: 'Поле заполнено неверно'
    }),

})

export const schema = yup.object().shape({
    email: yup.string().email('email не соответствует формату').required('заполните поле'),
    password: yup.string().required('заполните поле'),
    rememberMe: yup.boolean(),
});