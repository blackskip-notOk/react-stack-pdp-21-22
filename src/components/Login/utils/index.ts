import { SERVER_MESSAGES, SERVER_MESSAGES_DESCRIPTIONS } from './../../../constants/serverMessages';
import { RESULT_CODES } from './../../../constants/systemConstants';
import { LoginResponse } from "../../../api/loginApi/types";
import * as R from 'ramda';

export const loginErrorTransform = (data: LoginResponse | undefined, emptyValue = null): string | null => {
    if (!data || R.isEmpty(data) || data.resultCode === RESULT_CODES.success) {
        return emptyValue;
    };

    const { resultCode, messages } = data;

    if (resultCode === RESULT_CODES.error && messages.includes(SERVER_MESSAGES.WRONG_LOGIN)) {
        return SERVER_MESSAGES_DESCRIPTIONS.wrongLogin;
    };
    if (resultCode === RESULT_CODES.secure) {
        return messages.includes(SERVER_MESSAGES.MAX_ATTEMPT)
        ? SERVER_MESSAGES_DESCRIPTIONS.maxAttempt
        : SERVER_MESSAGES_DESCRIPTIONS.wrongLogin;
    };
    return emptyValue;
};

