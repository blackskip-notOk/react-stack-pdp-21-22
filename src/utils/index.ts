import { SyntheticEvent } from 'react';
import { ServerMessage, Description } from '../constants/serverMessages';
import { ReultCode } from '../constants/systemConstants';
import { FollowResponse, FollowResult, UsersRequestState } from '~/store/slices/usersSlice/types';
import { isEmpty } from 'ramda';
import { ClearObject } from './types';

export const preventDefault = (event: SyntheticEvent) => {
	event.preventDefault();
};

export const getFollowResult = (
	response: FollowResponse | undefined,
	user: string,
	isFollow: boolean,
): FollowResult => {
	if (!response) {
		return {
			isSuccess: false,
			message: Description.someError,
		};
	}

	const { messages, resultCode } = response;

	const successMessage = `${
		isFollow ? Description.successFollow : Description.successUnFollow
	} ${user}`;

	const unSuccessMessage = `${isFollow ? Description.alreadyFollow : Description.alreadyUnFollow}`;

	if (resultCode === ReultCode.success) {
		return {
			isSuccess: true,
			message: successMessage,
		};
	}
	if (resultCode === ReultCode.error) {
		const message = messages.join();

		const errorMessage =
			message === ServerMessage.alreadyUnfollow || message === ServerMessage.alreadyFollow
				? unSuccessMessage
				: Description.someError;

		return {
			isSuccess: false,
			message: errorMessage,
		};
	}
	return {
		isSuccess: false,
		message: Description.someError,
	};
};

export function* idGenerator() {
	let id = 0;
	while (true) yield ++id;
}

export const clearObject: ClearObject = (obj) => {
	const result = {} as Record<string, string>;

	for (const key in obj) {
		if (Object.hasOwn(obj, key)) {
			const param = obj[key];
			if (param !== null) {
				result[key] = param;
			}
		}
	}

	return isEmpty(result) ? null : result;
};

const getBooleanParam = (param: boolean | undefined): string | null => {
	if (param) {
		return '1';
	}

	return param === undefined ? null : '0';
};

export const searchParamsSerializer = (
	params: UsersRequestState,
): Record<string, string> | null => {
	const { page, count, term, friend } = params;

	const urlParams = {
		page: page === 1 ? null : String(page),
		count: count === 10 ? null : String(count),
		term: term ? term : null,
		friend: getBooleanParam(friend),
	};

	return clearObject(urlParams);
};

export const getSearchParamsFromUrl = (
	params: Record<string, string>,
): Partial<UsersRequestState> | null => {
	if (isEmpty(params)) {
		return null;
	}

	const { page, count, term, friend } = params;

	const getFriendParam = (param?: string) => {
		if (param === undefined) {
			return undefined;
		}

		return param === '1';
	};

	const result = {
		page: page ? Number(page) : undefined,
		count: count ? Number(count) : undefined,
		term: term || undefined,
		friend: getFriendParam(friend),
	};

	return result;
};

export const getFriendParam = (param: string): boolean | undefined => {
	if (!param || param === 'all') {
		return undefined;
	}

	return param === 'friends';
};
