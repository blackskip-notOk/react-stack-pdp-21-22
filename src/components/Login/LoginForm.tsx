import { Form, Input, Button, Checkbox } from 'antd';
import {SaveTwoTone} from "@ant-design/icons";
import {useForm} from "react-hook-form";
import {FormData} from "./types/LoginTypes";
import {yupResolver} from "@hookform/resolvers/yup";
import {loginFormSchema} from "../../validations/LoginValidation";

export const LoginForm = () => {
    const {register, handleSubmit, formState: {errors, touchedFields}
    } = useForm<FormData>({
        resolver: yupResolver(loginFormSchema)
    });

    const onSubmit = (data: FormData) => {
        console.log(data.email, data.password, data.rememberMe, data.captcha);
    }

    return (
        <Form
            layout='horizontal'
            name='loginForm'
            scrollToFirstError={true}
            size='large'
            onFinish={handleSubmit(onSubmit)}
            autoComplete='off'
        >
            <Form.Item label='email' name='email'>
                <Input {...register('email')}
                       type='text'
                       placeholder='email@xmail.com'
                />
            </Form.Item>
            <Form.Item label='password' name='password'>
                <Input {...register('password')}
                       type='password'
                       placeholder='minimum 4 symbols'
                />
            </Form.Item>
            <Form.Item label='rememberMe' name='rememberMe'>
                <Checkbox {...register('rememberMe')} type='checkbox'>
                    Remember me?
                </Checkbox>
            </Form.Item>
            <Form.Item label='captcha' name='captcha'>
                <Input {...register('captcha')}
                       type='text'
                       placeholder='enter symbols from image'
                />
            </Form.Item>
            <Button
                form='loginForm'
                htmlType='submit'
                icon={<SaveTwoTone />}
                // loading={{delay: 1000}}
                size='large'
                type='primary'
            >Submit</Button>
        </Form>
    )
}