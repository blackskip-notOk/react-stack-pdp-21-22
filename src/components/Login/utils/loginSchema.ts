import { useTranslation } from 'react-i18next';
import { object, boolean, string } from 'yup';

export const useLoginSchema = () => {
	const { t } = useTranslation();

	return object({
		email: string().defined().email(t('errors.invalidEmail')).required(t('errors.requiredEmail')),
		password: string().defined().required(t('errors.requiredPassword')),
		rememberMe: boolean().optional(),
		serverError: string(),
		captcha: string(),
	});
};
