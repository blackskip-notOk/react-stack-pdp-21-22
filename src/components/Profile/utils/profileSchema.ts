import { object, string } from 'yup';

export const profileStatusSchema = object({
	status: string().defined().optional().max(300, 'Maximum 300 symbols'),
});
