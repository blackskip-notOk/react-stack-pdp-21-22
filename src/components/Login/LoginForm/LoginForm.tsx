import { FC, useEffect, useReducer } from 'react';
import buttonStyles from '@/styles/Button.module.less';
import inputStyles from '@/styles/Input.module.less';
import { preventDefault } from '@/utils';
import { useNavigate } from 'react-router-dom';
import styles from '../Login.module.less';
import {
	Checkbox,
	FormControl,
	FormControlLabel,
	FormHelperText,
	IconButton,
	InputAdornment,
	InputLabel,
	OutlinedInput,
	TextField,
} from '@mui/material';
import { Visibility, VisibilityOff } from '@mui/icons-material';
import SendIcon from '@mui/icons-material/Send';
import { LoadingButton } from '@mui/lab';
import { useForm, Controller, SubmitHandler } from 'react-hook-form';
import { yupResolver } from '@hookform/resolvers/yup';
import { loginSchema } from '../utils/loginSchema';
import { Box } from '@mui/system';
import { isEmpty } from 'ramda';
import { NAVLINKS } from '@/constants/routerConstants';
import { LoginFormData } from '@/models/login/types';
import { LoginProps } from '../types';
import { useAppDispatch, useAppSelector } from '@/hooks/storeHooks';
import { getIsAuth } from '@/store/selectors/authSelectors';
import { getLoginState } from '@/store/selectors/loginSelector';
import { setLoginData, setLoginResponse } from '@/store/slices/loginSlice';
import { fetchCaptcha, loginApi } from '@/services/loginService';
import { getLoginResponse } from '@/services/helpers';

export const LoginForm: FC<LoginProps> = ({ setShowGreeting }) => {
	const navigate = useNavigate();

	const dispatch = useAppDispatch();
	const isAuth = useAppSelector(getIsAuth);
	const {
		data: { email, password, captcha, rememberMe },
		error: loginError,
		isNeedCaptcha,
		captchaUrl,
	} = useAppSelector(getLoginState);

	const [showPassword, toggleShowPassword] = useReducer((showPassword) => !showPassword, false);

	const [login, { isLoading: loginLoading, data: loginData, error: loginServerError }] =
		loginApi.useLoginMutation();

	useEffect(() => {
		if (isAuth) {
			setShowGreeting(true);
			navigate(NAVLINKS.HOME);
		}
	}, [isAuth]);

	useEffect(() => {
		if (loginData) {
			const loginResponse = getLoginResponse(loginData);

			dispatch(setLoginResponse(loginResponse));

			if (loginResponse.isNeedCaptcha) {
				dispatch(fetchCaptcha());
			}
		}
	}, [loginData]);

	const {
		control,
		handleSubmit,
		formState: { errors: formErrors },
	} = useForm<LoginFormData>({
		defaultValues: { email, password, rememberMe, captcha },
		resolver: yupResolver(loginSchema),
	});

	const onSubmit: SubmitHandler<LoginFormData> = async (data) => {
		dispatch(setLoginData(data));
		await login(data);
	};

	const handleGetCaptcha = () => {
		dispatch(fetchCaptcha());
	};

	return (
		<>
			<form onSubmit={handleSubmit(onSubmit)}>
				<Controller
					name='email'
					control={control}
					render={({ field }) => (
						<Box className={inputStyles.container}>
							<TextField
								{...field}
								id='email'
								label='Email'
								variant='outlined'
								helperText={formErrors.email?.message ?? ' '}
								error={!!formErrors.email || !!loginError}
								fullWidth
								color='success'
								margin='normal'
								focused
								placeholder='enter your email'
							/>
						</Box>
					)}
				/>
				<Controller
					name='password'
					control={control}
					render={({ field }) => (
						<FormControl
							color='success'
							variant='outlined'
							fullWidth
							focused
							className={inputStyles.container}
						>
							<InputLabel
								htmlFor='password'
								color='success'
								error={!!formErrors.password || !!loginError}
							>
								Password
							</InputLabel>
							<OutlinedInput
								{...field}
								id='password'
								label='Password'
								type={showPassword ? 'text' : 'password'}
								margin='dense'
								error={!!formErrors.password || !!loginError}
								placeholder='enter your password'
								endAdornment={
									<InputAdornment position='end'>
										<IconButton
											aria-label='toggle password visibility'
											onClick={toggleShowPassword}
											onMouseDown={preventDefault}
											edge='end'
										>
											{showPassword ? (
												<VisibilityOff color='success' />
											) : (
												<Visibility color='success' />
											)}
										</IconButton>
									</InputAdornment>
								}
							/>
							<FormHelperText id='passwordError' error={!!formErrors.password}>
								{formErrors.password?.message ?? ' '}
							</FormHelperText>
						</FormControl>
					)}
				/>
				<Controller
					name='rememberMe'
					control={control}
					defaultValue={true}
					render={({ field }) => (
						<FormControlLabel
							className={styles.checkBox}
							control={
								<Checkbox
									{...field}
									defaultChecked
									sx={{ '& .MuiSvgIcon-root': { fontSize: 28 } }}
									color='success'
									inputProps={{
										'aria-label': 'remember me checkbox',
									}}
								/>
							}
							label='Remember me?'
							labelPlacement='start'
						/>
					)}
				/>{' '}
				{isNeedCaptcha && (
					<Controller
						name='captcha'
						control={control}
						render={({ field }) => (
							<Box className={inputStyles.container}>
								<TextField
									{...field}
									id='captcha'
									label='Captcha'
									variant='outlined'
									required
									helperText={formErrors.captcha?.message ?? ' '}
									error={!!formErrors.captcha}
									fullWidth
									color='success'
									margin='normal'
									focused
									placeholder='enter text from picture'
								/>
							</Box>
						)}
					/>
				)}
				{loginServerError && (
					<Box className={styles.autorizationError}>{loginServerError.message}</Box>
				)}
				{loginError && <Box className={styles.autorizationError}>{loginError}</Box>}
				<Box className={buttonStyles.buttonContainer}>
					<LoadingButton
						size='large'
						endIcon={<SendIcon />}
						type='submit'
						loading={loginLoading}
						disabled={loginLoading || !isEmpty(formErrors)}
						loadingPosition='end'
						variant='contained'
						color='success'
					>
						Логин
					</LoadingButton>
				</Box>
			</form>
			{isNeedCaptcha && (
				<div className={buttonStyles.buttonContainer}>
					<LoadingButton
						onClick={handleGetCaptcha}
						size='large'
						endIcon={<SendIcon />}
						loading={loginLoading}
						disabled={loginLoading || !isEmpty(formErrors)}
						loadingPosition='end'
						variant='contained'
						color='success'
					>
						Другая картинка
					</LoadingButton>
					{captchaUrl && <img src={captchaUrl} alt='captcha' className={styles.img} />}
				</div>
			)}
		</>
	);
};
