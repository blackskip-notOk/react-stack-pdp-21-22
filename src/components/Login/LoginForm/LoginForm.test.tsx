import { fireEvent, screen } from '@testing-library/react';
import { renderWithProviders, setup } from '~/utils/testUtils';
import { LoginForm } from './LoginForm';

describe('LoginForm.tsx', () => {
	const setShowGreeting = vi.fn();
	describe('validation login form', () => {
		test('if click on login button without filled login/password, should show validation messages', async () => {
			renderWithProviders(<LoginForm setShowGreeting={setShowGreeting} />);

			fireEvent.click(screen.getByRole('button', { name: /login/i }));
			expect(await screen.findByText(/email is required/i)).toBeInTheDocument();
			expect(await screen.findByText(/password is required/i)).toBeInTheDocument();
			expect(await screen.findByRole('button', { name: /login/i })).toBeDisabled();
		});
		test('if user typed incorrect email, should show validation error', async () => {
			const { user } = setup(<LoginForm setShowGreeting={setShowGreeting} />);
			const emailInput = screen.getByRole('textbox', { name: /email/i });
			await user.click(emailInput);
			await user.type(emailInput, 'testEmail');

			expect(emailInput).toHaveValue('testEmail');
			// expect(await screen.findByText(/enter valid email/i)).toBeInTheDocument();
			// expect(container).toHaveTextContent(/errorEmail/i); // TODO research how to find a correct error message
		});
	});

	// describe('dispatching actions to a redux store', () => {
	// test('if user is auth, should set login request data to initial', () => {});
	// });
});
