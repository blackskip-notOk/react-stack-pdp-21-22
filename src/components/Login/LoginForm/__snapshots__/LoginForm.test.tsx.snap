// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="MuiBox-root css-0"
  >
    <div
      className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorSuccess Mui-focused css-3v5nb6-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={true}
        htmlFor="email"
        id="email-label"
      >
        Email
      </label>
      <div
        className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorSuccess MuiInputBase-fullWidth Mui-focused MuiInputBase-formControl css-bz6wus-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-describedby="email-helper-text"
          aria-invalid={false}
          autoFocus={false}
          className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id="email"
          name="email"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="enter your email"
          required={false}
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-1in441m"
          >
            <span>
              Email
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        className="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained Mui-focused css-1wc848c-MuiFormHelperText-root"
        id="email-helper-text"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </p>
    </div>
  </div>
  <div
    className="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
  >
    <label
      className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorSuccess Mui-focused css-3v5nb6-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={true}
      htmlFor="password"
    >
      Password
    </label>
    <div
      className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorSuccess Mui-focused MuiInputBase-formControl MuiInputBase-adornedEnd css-sqs87j-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id="password"
        name="password"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="enter your password"
        required={false}
        type="password"
        value=""
      />
      <div
        className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
      >
        <button
          aria-label="toggle password visibility"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-colorSuccess MuiSvgIcon-fontSizeMedium css-4b67si-MuiSvgIcon-root"
            data-testid="VisibilityIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
            />
          </svg>
        </button>
      </div>
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-1in441m"
        >
          <span>
            Password
          </span>
        </legend>
      </fieldset>
    </div>
    <p
      className="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained Mui-focused css-1wc848c-MuiFormHelperText-root"
      id="passwordError"
    >
      <span
        className="notranslate"
      >
        ​
      </span>
    </p>
  </div>
  <label
    className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart css-1f2kvjf-MuiFormControlLabel-root"
  >
    <span
      className="MuiCheckbox-root MuiCheckbox-colorSuccess MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorSuccess PrivateSwitchBase-root Mui-checked css-191jftd-MuiButtonBase-root-MuiCheckbox-root"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
    >
      <input
        aria-label="remember me checkbox"
        className="PrivateSwitchBase-input css-1m9pwf3"
        data-indeterminate={false}
        defaultChecked={true}
        name="rememberMe"
        onChange={[Function]}
        type="checkbox"
        value={true}
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CheckBoxIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
    </span>
    <span
      className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
    >
      Remember me?
    </span>
  </label>
   
  <div
    className="MuiBox-root css-0"
  >
    <button
      className="MuiLoadingButton-root MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root css-8hdand-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
      disabled={false}
      id=":r1:"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      Логин
      <span
        className="MuiButton-endIcon MuiButton-iconSizeLarge css-z4rqyd-MuiButton-endIcon"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="SendIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M2.01 21 23 12 2.01 3 2 10l15 2-15 2z"
          />
        </svg>
      </span>
    </button>
  </div>
</form>
`;
